thumbor/src/thumbor_/handlers/upload_handler.py

```diff
@@ -1,40 +1,41 @@
-from s3_store.s3_storage import Storage
+import os
 from io import BytesIO
-import loaders.bucket_details as bucket_details
-from botocore.config import Config as BotoConfig
-import boto3
+
 import tornado.web
 
+from thumbor_.s3_store.s3_storage import Storage
 
 
 class UploadHandler(tornado.web.RequestHandler):
-    def initialize(self, bucket_name):
-        self.bucket_name = bucket_name
+    def initialize(self):
+        self.bucket_name = os.environ.get("BUCKET_NAME")
         self.storage = Storage(context=self)
-        self.s3 = boto3.client(
-            's3',
-            region_name="eu-west-2",
-            endpoint_url=bucket_details.ep_url,
-            aws_access_key_id=bucket_details.key_id,
-            aws_secret_access_key=bucket_details.access_key,
-            config=BotoConfig(signature_version="s3v4"),
-        )
 
-    def put(self):
-        fileinfo = self.request.files['media'][0]
-        filename = fileinfo['filename']
-        filebody = fileinfo['body']
+    async def put(self):
+        fileinfo = self.request.files["media"][0]
+        filename = fileinfo["filename"]
+        filebody = fileinfo["body"]
+        lifespan = self.get_query_argument("lifespan", default="180")
         try:
-            path = self.storage.put(filename, BytesIO(filebody))
-            self.write({'status': 'success', 'message': f'File {path} uploaded successfully to S3'})
+            path = await self.storage.put(filename, BytesIO(filebody), lifespan)
+            self.write(
+                {
+                    "status": "success",
+                    "message": f"File {path} uploaded successfully to S3",
+                }
+            )
         except Exception as e:
-            self.write({'status': 'error', 'message': str(e)})
-
+            self.write({"status": "error", "message": str(e)})
 
-    def delete(self):
-        filename = self.get_argument('filename')
+    async def delete(self):
+        filename = self.get_argument("filename")
         try:
-            path = self.storage.delete(filename)
-            self.write({'status': 'success', 'message': f'File {path} deleted successfully from S3'})
+            path = await self.storage.delete(filename)
+            self.write(
+                {
+                    "status": "success",
+                    "message": f"File {path} deleted successfully from S3",
+                }
+            )
         except Exception as e:
-            self.write({'status': 'error', 'message': str(e)})
\ No newline at end of file
+            self.write({"status": "error", "message": str(e)})
```

Comment:
Could we do something like `self.set_status(...)` here to make the response have an appropriate http error code?

Reply:
if upload fails client gets appropriate status code